docs/faq.md

```diff
@@ -402,3 +402,15 @@ To fix student scores without requiring them to answer another question you can:
 4. Upload the new scores with the "Upload new total scores" button on the "Uploads" tab.
 
 Changing total scores via CSV download/upload should only be done after the assessment is over and students are not working on it anymore, to avoid any risk of overwriting scores while students are answering more questions.
+
+## I updated my question, but students are still seeing the old version. How can I get students to see the new version?
+
+Some changes in question files do not take effect for students that have already started working on a variant. These include, but are not limited to: changes in the value of correct answers, added or removed graded elements, changes in element type (e.g., replacing string input with integer input), some other changes in element settings (e.g., multiple choice answers), changes in file names for gradable files, or updates to the `generate` function in `server.py`.
```

Comment:
This has bitten me a couple of times in the past, but as I'm reading the code this may no longer be the case. Note that `params._required_file_names` is updated in the prepare stage in pl-file-editor/pl-file-upload/pl-rich-text-editor and in the initialization of questions that use workspace graded files, though at this point it doesn't seem like this value is used anywhere. That said, at the very least, changing the gradedFiles in workspace options will cause issues:

https://github.com/PrairieLearn/PrairieLearn/blob/db2621c211c44b33e51eb5820f2a1d9c9773edc8/apps/prairielearn/elements/pl-workspace/pl-workspace.py#L23-L37

In any case, the following use case I just tested can cause an issue:
* A question has a pl-file-editor with file name a.txt
* The student opens a variant and starts to write text in a.txt
* The instructor changes the pl-file-editor to use file name b.txt
* The student submits the code with the old name (without refreshing)

The scenario above would cause the submission to be sent with a.txt, which in the new version has no element that will take care of it, since the input name is based on a hash of the file name. This means that the student's submission will be "lost" (though technically still available in the raw submitted answer), and the student will get an error saying b.txt has no submission.

Reply:
TL;DR: the file name is what determines the input name, so changing it is equivalent to changing answers-name, which is known to cause problems.